src/libtiled/templatemanager.cpp

```diff
@@ -61,14 +65,28 @@ ObjectTemplate *TemplateManager::loadObjectTemplate(const QString &fileName, QSt
     if (!objectTemplate) {
         auto newTemplate = readObjectTemplate(fileName, error);
 
-        // This instance will not have an object. It is used to detect broken
-        // template references.
-        if (!newTemplate)
+        if (newTemplate) {
+            mWatcher->addPath(fileName);
+        } else {
+            // This instance will not have an object. It is used to detect broken
+            // template references.
             newTemplate = std::make_unique<ObjectTemplate>(fileName);
+        }
 
         objectTemplate = newTemplate.get();
         mObjectTemplates.insert(fileName, newTemplate.release());
     }
 
     return objectTemplate;
 }
+
+void TemplateManager::fileChanged(const QString &fileName)
+{
+    ObjectTemplate *objectTemplate = findObjectTemplate(fileName);
+
+    // Most likely the file was removed.
+    if (!objectTemplate)
```

Comment:
File removal does not remove the template from `mObjectTemplates`.

Actually at the moment nothing does, so this check shouldn't be needed... but it doesn't hurt either.

Reply:
I think it's prudent to keep the check, just in case something else changes in the rest of the codebase.

Reply:
Yeah, like I said it doesn't hurt, but the misleading comment should be removed in any case. :-)